lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1670,6 +1670,147 @@ def test_certificate_generation_for_students(self):
             result
         )
 
+    def test_certificate_generation_for_students_semantic_task_input_all_whitelisted(self):
```

Comment:
these test cases all seem really similar — can you abstract out some methods or combine these cases with `ddt`?